constructor/header.sh

```diff
@@ -671,11 +671,22 @@ if [ "$BATCH" = "0" ]; then
             *) "$PREFIX/bin/python" -m conda init ;;
         esac
         if [ -f "$PREFIX/bin/mamba" ]; then
```

Comment:
Seeing this code now, I think it might result into double initialization, right? Once with conda, once with mamba. Are those compatible or should they be exclusive?

Reply:
I think conda's initialization is overridden by mamba's; so it works, but it might not be totally sane. I guess we can keep the same behavior when 1.x is used unchanged, and see potential adaptations for 2.x.

What do you think? Is there a way to test that this works with both versions?

Reply:
We'll need to add cases for it in the `examples/` directory. Maybe you can repurpose the `Miniforge` example (which might need to get synced up with current Miniforge scripts anyway), and then parametrize with some environment variables as we are doing in some other tests. Check for `os.environ` in the `construct.yaml` templates and also `monkeypatch.setenv` calls in `tests/test_examples.py`.